### PR TITLE
feat: get started guide for the Mender Gateway

### PR DESCRIPTION
**Pull request by tranchitella**

### CODE REVIEW

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+The Mender Gateway devices have a special inventory attribute, `mender_is_gateway=true`.
+
+Thanks to this, you can filter these devices and eventually create a dynamic group, as shown below:
+
+![Mender Gateway inventory attribute](mender-gateway-inventory.png)
+
+
+## Obtain the Mender Gateway's IP address 
+
+Before connecting other devices to it, we need to obtain the Mender Gateway's IP address we will point them to.
```
Comment (merlin-northern) [inline 01.Get-started/06.Mender-Gateway/docs.md:148]: somehow this sentence could be better, maybe "_In order to connect devices to the Mender Gateway, we need to obtain its IP address._" or something else.

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+> Apr 08 06:10:44 raspberrypi mender-gateway[12374]: time="2022-04-08T06:10:44+02:00" level=info msg="starting mender-gateway" file=main.go func=main.doMain line=96
+> Apr 08 06:10:44 raspberrypi mender-gateway[12374]: time="2022-04-08T06:10:44+02:00" level=info msg="creating proxy with url https://hosted.mender.io, insecure skip verify: false" file=proxy.go func=proxy.NewProxy line=47
+> Apr 08 06:10:44 raspberrypi mender-gateway[12374]: time="2022-04-08T06:10:44+02:00" level=info msg="proxy scheme: https, host: hosted.mender.io" file=proxy.go func=proxy.NewProxy line=54
+> Apr 08 06:10:44 raspberrypi mender-gateway[12374]: time="2022-04-08T06:10:44+02:00" level=info msg="created client with base url https://hosted.mender.io, insecure skip verify: false" file=client.go func=mender.NewClient line=49
+> Apr 08 06:10:44 raspberrypi mender-gateway[12374]: time="2022-04-08T06:10:44+02:00" level=info msg=running... file=server.go func="server.(*Server).Run" line=75
+> ```
+
+You can start, stop or restart the service using `systemctl`:
```
Comment (merlin-northern) [inline 01.Get-started/06.Mender-Gateway/docs.md:110]: I think this is a series consisting of "start", "stop", "or restart", the third comma should be there (right @michaelatmender ? I am not an expert here.) it should be: "You can start, stop_,_ or restart the service ..."

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+
+```bash
+sudo systemctl restart mender-gateway
+```
+
+To inspect the logs, you can use `journalctl`:
```
Comment (merlin-northern) [inline 01.Get-started/06.Mender-Gateway/docs.md:116]: I am not convinced we need comma here.

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+	mendersoftware/mender-client-qemu
+```
+
+!!! Be aware that the environment variable `SERVER_IP` must be set to the Raspberry PI's local network
+!!! IP address. The Docker container running the Mender Client will automatically add a line in `/etc/hosts`
+!!! to make `gateway.docker.mender.io` point to the specified IP address.
+
+![Pending device](pending-device.png)
+
+The device will connect to the Mender Gateway, which will forward the API calls to Hosted Mender or
```
Comment (merlin-northern) [inline 01.Get-started/06.Mender-Gateway/docs.md:190]: no comma should be here as far as I know.

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+your upstream Mender Server. The virtual device will show up in the Mender UI as a new pending device.
+You can accept your devices as usual to authorize the connection to the Mender Server.
+
+![Accepted devices](accepted-devices.png)
+
+At this point, the device will act as the communication between it and the Mender Server would be
```
Comment (merlin-northern) [inline 01.Get-started/06.Mender-Gateway/docs.md:196]: something is missing in this sentence. perhaps: "At this point, the device will act as the communication between it and the Mender Server would be" -> "At this point, the device will act as _if_ the communication between it and the Mender Server _is_"

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+
+![Remote terminal](remote-terminal.png)
+
+To prove the device has no direct connection to the Mender Server, you could add an entry
+in your `/etc/hosts` file to point the Mender Server's domain name to a wrong IP address:
+
+```bash
+echo "0.1.1.1 hosted.mender.io" | sudo tee -a /etc/hosts
+curl hosted.mender.io
+```
```
Comment (merlin-northern) [inline 01.Get-started/06.Mender-Gateway/docs.md:211]: maybe a more convincing example would be to use packet filter to drop all outgoing packets to the hosted.mender.io?

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
+---
+title: Mender Gateway
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply 01.Get-started/06.Mender-Gateway/docs.md:2]: Added

File: 01.Get-started/06.Mender-Gateway/docs.md (hunk 6 above)
Comment (tranchitella) [reply 01.Get-started/06.Mender-Gateway/docs.md:211]: Maybe; on the other hand, I'd keep it simple and leave the iptables exercise to the reader.

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 8]
```diff
@@ -0,0 +1,231 @@
+---
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply 01.Get-started/06.Mender-Gateway/docs.md:1]: All the pictures are screenshot from my Mac Book, exported (again) with Gimp with the maximum compression.
We could switch from PNG to JPEG adding compression. Shall we do it?

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+
+## Installation
+
+The first step is installing the Mender Gateway into your Raspberry Pi device.
+
+You can use the "Connect a device" dropdown in the Mender UI and select the
+"Promote a device to gateway" menu item to generate a snippet to install easily
+the Mender Gateway package into your device.
+
+![Promote a device to gateway](promote-device-to-gateway.png)
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply 01.Get-started/06.Mender-Gateway/docs.md:34]: See the other comment about the screenshots.

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 10]
```diff
@@ -0,0 +1,231 @@
...
+    label: tutorial
+---
+
+!!!!! Mender Gateway is only available in the Mender Enterprise plan.
+!!!!! See [the Mender features page](https://mender.io/plans/features?target=_blank)
+!!!!! for an overview of all Mender plans and features.
+
+This tutorial will walk you through setting up a
+[Mender Gateway](../../08.Server-integration/04.Mender-Gateway/) in your local network
+to connect one local device running Mender to an upstream Mender server. To simulate a
```
[the remark this answers is not in the document]
Comment (TheYoctoJester) [reply 01.Get-started/06.Mender-Gateway/docs.md]: ... especially as its an upper case S three lines down.

File: 01.Get-started/06.Mender-Gateway/docs.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (merlin-northern) [reply 01.Get-started/06.Mender-Gateway/docs.md:1]: I did not mean the size in bytes :) but the resolution, perhaps re-size and crop? it is a nitpick, I do not want to cause you more work.

**Review by TheYoctoJester — APPROVED** [review]

**Comment by tranchitella** [comment]
@TheYoctoJester I got your "review comments" just now, it looks like they were pending on your end.
I will address them before merging.

**Comment by TheYoctoJester** [comment]
@tranchitella strange. But yes, thanks!

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+segregated network and demonstrate the use-case, we will make it impossible for the
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply 01.Get-started/06.Mender-Gateway/docs.md:15]: As it seems we prefer to have as few commas as possible, I'm removing the extra ones in the lists.

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 12]
```diff
@@ -0,0 +1,231 @@
...
+```
+
+The `JWT_TOKEN` environment variable contains a valid Hosted Mender JWT token, which the script
+will use to authenticate to the server when downloading the Mender Gateway binaries.
+
+
+## Configuration
+
+When installing the Mender Gateway in demo mode (see the `--demo` flag in the snippet above),
+the package automatically configures it to connect to Hosted Mender, proxying both device APIs
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply 01.Get-started/06.Mender-Gateway/docs.md:51]: Right, let's use "forwarding".

File: 01.Get-started/06.Mender-Gateway/docs.md [hunk 13]
```diff
@@ -0,0 +1,231 @@
...
+to connect one local device running Mender to an upstream Mender Server. To simulate a
+segregated network and demonstrate the use-case, we will make it impossible for the
+local device to connect to the Mender Server directly.
+
+
+## Prerequisites
+
+To follow this tutorial, you will need to
+[prepare a Raspberry Pi device](../01.Preparation/01.Prepare-a-Raspberry-Pi-device/)
+and connect it to Hosted Mender or your Mender Server.
```
[the remark this answers is not in the document]
Comment (tranchitella) [reply 01.Get-started/06.Mender-Gateway/docs.md:23]: You don't need docker, we run the Mender Gateway as a systemd service.

File: 01.Get-started/06.Mender-Gateway/docs.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (tranchitella) [reply 01.Get-started/06.Mender-Gateway/docs.md:1]: Let's keep the screenshots as they are, then.

File: 01.Get-started/06.Mender-Gateway/docs.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (TheYoctoJester) [reply 01.Get-started/06.Mender-Gateway/docs.md:23]: You don't need it on the device, but you need it on the host that runs the virtual device. Therefore it is a prerequisite for the getting started guide as a whole.

**Comment by oleorhagen** [comment]
@mender-test-bot cherry-pick to: `hosted`

**Pull request merged**